lib/notifications/display.dart

```diff
@@ -92,7 +93,29 @@ class NotificationDisplayManager {
   static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
     assert(debugLog('notif message content: ${data.content}'));
     final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
-    final title = switch (data.recipient) {
+    final groupKey = _groupKey(data);
+    final conversationKey = _conversationKey(data, groupKey);
+
+    final oldMessagingStyle = await ZulipBinding.instance.androidNotificationHost
+      .getActiveNotificationMessagingStyleByTag(conversationKey);
+
+    final MessagingStyle messagingStyle = oldMessagingStyle != null
+      ? MessagingStyle(
+          user: oldMessagingStyle.user,
+          messages: oldMessagingStyle.messages?.toList() ?? [], // Clone a fixed-length list
+          isGroupConversation: oldMessagingStyle.isGroupConversation)
+      : MessagingStyle(
+          user: Person(
+            key: data.userId.toString(),
+            name: 'You'), // TODO(i18n)
+          messages: [],
+          isGroupConversation: switch (data.recipient) {
+            FcmMessageStreamRecipient() => true,
+            FcmMessageDmRecipient(:var allRecipientIds) when allRecipientIds.length > 2 => true,
+            FcmMessageDmRecipient() => false,
+        });
+
+    messagingStyle.conversationTitle = switch (data.recipient) {
```

Comment:
There's a comment above this in the corresponding zulip-mobile code, about the title changing between messages in a conversation, which I think is still relevant.